README.md

```diff
@@ -28,13 +27,298 @@ pip install -U pyinstaller-hooks-contrib
+
+Tests currently all live in [src/_pyinstaller_hooks_contrib/tests/test_libraries.py](../master/src/_pyinstaller_hooks_contrib/tests/test_libraries.py).
+Navigate there and add something like the following, replacing all occurrences of `foo` with the real name of the library.
+(Note where you put it in that file doesn't matter.)
+
+```python
+@importorskip('foo')
+def test_foo(pyi_builder):
+    pyi_builder.test_source("""
+
+	# Your test here!
+        import foo
+
+	foo.something_fooey()
+
+    """)
+```
+
+If need be, you can specify version constraints using `@importorskip('foo', minversion="1.2.3")`
+if the library has changed significantly between versions.
+
+
+#### Run the test locally
+
+Running our full test suite is not recommended as it will spend a very long time testing code which you have not touched.
+Instead, run tests individually using either the `-k` option to search for test names:
+
+```
+pytest -k test_foo
+```
+
+Or using full paths:
+```
+pytest src/_pyinstaller_hooks_contrib/tests/test_libraries.py::test_foo
+```
+
+
+#### Run the test on CI/CD
+
+To test hooks on all platforms we use Github's continuous integration (CI/CD).
+Our CI/CD is a bit unusual in that it's triggered manually and takes arguments
+which limit which tests are run.
+This is for the same reason we filter tests when running locally -
+the full test suite runs like a dead snail on ketamine.
+
+First push the changes you've made so far.
+
+```commandline
+git push --set-upstream origin hook-for-foo
+```
+
+Replace *billy-the-buffalo* with your Github username in the following url then open it.
+It should take you to the `oneshot-test` actions workflow on your fork.
+You may be asked if you want to enable actions on your fork - say yes.
+```
+https://github.com/billy-the-buffalo/pyinstaller-hooks-contrib/actions/workflows/oneshot-test.yml
+```
+
+Find the **Run workflow** button and click on it.
+A dialog should appear containing one drop-down menu and 5 line-edit fields.
+This dialog is where you specify what to test and which platforms and Python versions to test on.
+Its fields are as follows:
+1.  A branch to run from. Set this to the branch which you are using (e.g. ``hook-for-foo``),
+2.  Which package(s) to install and their version(s).
+    Which packages to test are inferred from which packages are installed.
+    * Set to `foo` to test the latest version of `foo`,
+    * Set to `foo==1.2, foo==2.3` (note the comma) to test two different versions of `foo` in separate jobs,
+    * Set to `foo bar` (note the lack of a comma) to test `foo` and `bar` in the same job,
+3.  Which OS or OSs to run on
+    * Set to `ubuntu` to test only `ubuntu`,
+    * Set to `ubuntu, macos, windows` (order is unimportant) to test all three OSs.
+4.  Which Python version(s) to run on
+    * Set to `3.9` to test only Python 3.9,
+    * Set to `3.6, 3.7, 3.8, 3.9` to test all currently supported version of Python.
+5.  The final two options can generally be left alone.
```

Comment:
Since this is a step-by-step guide, it would be helpful to say before the step that you don't need to follow it. I was following along and got confused about how I was supposed to make the decisions for steps 3 and 4.

Reply:
What do you mean by you don't need to follow it? Do you just mean that the defaults will work or something else?

Reply:
I think I misread this the first time. It's worded fine.